### PR TITLE
WIP: [form-builder] Add support for block objects in the new PTE implement…

### PR DESCRIPTION
**Pull request by saasen**
WIP for custom block preview + editing in the new PTE implementation.

### CODE REVIEW

File: packages/@sanity/form-builder/src/inputs/PortableText/BlockRendering/BlockPreview.tsx [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+      icon: TrashIcon,
+      name: 'delete',
+      color: 'danger'
+    })
+  }
+
+  const handleHeaderMenuAction = (item: DropDownButtonItem): void => {
+    // const {node, editor} = this.props
+    if (item.name === 'delete') {
+      // onClickingDelete([unset(path)])
```
Comment (saasen) [inline packages/@sanity/form-builder/src/inputs/PortableText/BlockRendering/BlockPreview.tsx:94]: We can't use the `handleChange` as the delete action as it expects a type `PatchEvent`, not a `UnsetPatch`. Any clever ideas here, @skogsmaskin?

Comment (skogsmaskin) [reply packages/@sanity/form-builder/src/inputs/PortableText/BlockRendering/BlockPreview.tsx:94]: You can do `PatchEvent.from([unset([...])])`

Comment (saasen) [reply packages/@sanity/form-builder/src/inputs/PortableText/BlockRendering/BlockPreview.tsx:94]: Just what kind of reply I wanted.

File: packages/@sanity/form-builder/src/inputs/ArrayInput/typedefs.ts [hunk 2]
```diff
@@ -1,12 +1,14 @@
...
+export type ValidEditModalValues = 'modal' | 'fullscreen' | string
+
 export type ArrayType = Type & {
   name: string
   title: string
   description: string
   readOnly: boolean | null
   options: {
-    editModal?: 'fold' | 'modal' | string
+    editModal?: ValidEditModalValues
```
Comment (skogsmaskin) [inline packages/@sanity/form-builder/src/inputs/ArrayInput/typedefs.ts:11]: Couldn't this type just be named `ModalType` ?

Comment (saasen) [reply packages/@sanity/form-builder/src/inputs/ArrayInput/typedefs.ts:11]: Yes, I agree. Will rename!

**Comment by saasen** [comment]
Done, @skogsmaskin! 👏

**Pull request merged**